### PR TITLE
feat(http): add blocklist manager

### PR DESCRIPTION
**Pull request by Oleksiivanov**
## Description

Added http blocklist

## Related issues

closes #
https://github.com/camunda/connectors/issues/1286

### CODE REVIEW

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/HttpBlocklistManager.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+public class HttpBlocklistManager {
+
+  private static final String BLOCK_PREFIX = "CONNECTORS_HTTP_BLOCK_";
+  private final Map<BlockType, Set<String>> blocklist = new HashMap<>();
+  private final Map<String, Pattern> regexCache = new HashMap<>();
+
+  enum BlockType {
+    URL("URL"),
+    URL_PARTIAL_MATCH("URLPARTIAL"),
```
Comment (sbuettner) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/HttpBlocklistManager.java:36]: URL should support partial matches as well. Maybe just have URL and check with `contains`?

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/HttpBlocklistManager.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  public void validateUrlAgainstBlocklist(GenericUrl url) {
+    blocklist.forEach(
+        (blockType, values) -> {
+          for (String value : values) {
+            switch (blockType) {
+              case URL -> checkUrl(value, url.build());
+              case URL_PARTIAL_MATCH -> checkUrlPartialMatch(value, url.build());
+              case DOMAIN -> checkDomain(value, url.getHost());
+              case PORT -> checkPort(value, String.valueOf(url.getPort()));
+              case IP -> checkIP(value, url.getHost());
```
Comment (sbuettner) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/HttpBlocklistManager.java:92]: The docs say 
> Returns the host, for example "www.google.com".
and therefore its the hostname not the actual host IP. 

Either validate against the resolved IP or remove the blocktype for now.

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/HttpBlocklistManager.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        String blockTypeStr = entry.getKey().substring(BLOCK_PREFIX.length()).split("_")[0];
+        Optional.ofNullable(BlockType.valueOfName(blockTypeStr))
+            .ifPresent(
+                blockType ->
+                    blocklist
+                        .computeIfAbsent(blockType, k -> new HashSet<>())
+                        .add(entry.getValue()));
+      }
+    }
+  }
```
Comment (sbuettner) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/HttpBlocklistManager.java:81]: This method is untested. Please add a test.

Also please store the name of the block and make it available when trowing an error.
Example: `CONNECTORS_HTTP_BLOCK_URL_GOOGLE_META=metaData`
Name: `GOOGLE_META`.

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/DefaultHttpBlocklistManager.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+  private static final Logger logger = LoggerFactory.getLogger(DefaultHttpBlocklistManager.class);
+  private static final String BLOCK_PREFIX = "CONNECTORS_HTTP_BLOCK_";
+
+  private final Map<BlockType, Map<String, Set<String>>> blockNameList = new HashMap<>();
+  private final Map<String, Pattern> regexCache = new HashMap<>();
+
+  enum BlockType {
+    URL("URL"),
+    DOMAIN("DOMAIN"),
```
Comment (sbuettner) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/DefaultHttpBlocklistManager.java:39]: Isnt the domain also part of the url?

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/DefaultHttpBlocklistManager.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                for (String value : values) {
+                  switch (blockType) {
+                    case URL -> checkUrl(value, url.build(), blockName);
+                    case DOMAIN -> checkDomain(value, url.getHost(), blockName);
+                    case PORT -> checkPort(value, String.valueOf(url.getPort()), blockName);
+                    case REGEX -> checkRegex(value, url.build(), blockName);
+                  }
+                }
+              });
+        });
```
Comment (sbuettner) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/DefaultHttpBlocklistManager.java:114]: Wouldnt it be simpler to use a sealed record class for each block?

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/block/PortBlock.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(value, blockName);
+  }
+
+  @Override
+  public String toString() {
+    return "PortBlock[" + "value=" + value + ", " + "blockName=" + blockName + ']';
+  }
```
Comment (sbuettner) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/block/PortBlock.java:111]: Could you use records for the block implementations?

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/DefaultHttpBlocklistManager.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import io.camunda.connector.http.base.blocklist.block.Block;
+import io.camunda.connector.http.base.blocklist.factory.BlockFactory;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class DefaultHttpBlocklistManager implements HttpBlockListManager {
+
+  private static final String BLOCK_PREFIX = "CONNECTORS_HTTP_BLOCK_";
```
Comment (sbuettner) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/DefaultHttpBlocklistManager.java:29]: Lets use `CAMUNDA_CONNECTOR_HTTP_` as the prefix.

Comment (sbuettner) [reply connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/DefaultHttpBlocklistManager.java]: Ah, sorry misunderstanding. Yes, I meant: `CAMUNDA_CONNECTOR_HTTP_BLOCK_URL_NAME`

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/block/PortBlock.java (hunk 6 above)
Comment (sbuettner) [reply connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/block/PortBlock.java:111]: You can have a constructor also for records if thats what you want: https://www.baeldung.com/java-records-custom-constructor

Comment (sbuettner) [reply connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/block/PortBlock.java:111]: First of all we didnt specify we need regex so we are overdoing this a little bit 😄 

You could also require to have a pattern in the constructor:  MyRecord(String name, Pattern regex) and create a static factory method that creates the instance.

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/factory/BlockFactory.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+public final class BlockFactory {
+  private static final Logger logger = LoggerFactory.getLogger(BlockFactory.class);
+
+  private static final String URL_TYPE = "URL";
+  private static final String PORT_TYPE = "PORT";
+  private static final String REGEX_TYPE = "REGEX";
+
+  private BlockFactory() {}
+
+  public static Optional<Block> createBlock(String blockType, String blockRule, String blockName) {
```
Comment (sbuettner) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/factory/BlockFactory.java:37]: Lets simplify: Either throw exception or return a block and not an Optional.

**Review by sbuettner — DISMISSED** [review]
Looks good! 

Only a minor comment

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/block/PortBlock.java (hunk 6 above)
Comment (sbuettner) [reply connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/block/PortBlock.java]: Why would you need the `String value` in the record?

**Comment by sbuettner** [comment]
@Oleksiivanov Could you try to deploy this new version of the http connector as part of the proxy deployment?

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/DefaultHttpBlocklistManager.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        .map(this::createBlockFromEnvEntry)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  private Block createBlockFromEnvEntry(Map.Entry<String, String> entry) {
+    var remaining = entry.getKey().substring(BLOCK_PREFIX.length());
+    var parts = remaining.split("_", 2);
+    var blockType = parts[0];
+    var blockName = parts.length > 1 ? parts[1] : "";
```
Comment (igpetrov) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/DefaultHttpBlocklistManager.java:57]: Suggestion to at least leave comments what those magic numbers are, maybe with examples.

**Review by igpetrov — DISMISSED** [review]
LGTM, minor comment. Feel free to address it in the other PR if you're in hurry with it.

**Review by igpetrov — APPROVED** [review]
LGTM, thank you

**Pull request merged**